dev/i18n/check_translations_completeness.py

```diff
@@ -581,12 +614,84 @@ def natural_key_sort(obj):
             return obj
 
         lang_data = natural_key_sort(lang_data)
+
+        if dry_run:
+            console.print(f"[blue][DRY RUN] Would add missing translations to {lang_path}[/blue]")
+            return
+
         lang_path.parent.mkdir(parents=True, exist_ok=True)
         with open(lang_path, "w", encoding="utf-8") as f:
             json.dump(lang_data, f, ensure_ascii=False, indent=2)
             f.write("\n")  # Ensure newline at the end of the file
         console.print(f"[green]Added missing translations to {lang_path}[/green]")
 
 
+def remove_extra_translations(
+    language: str, summary: dict[str, LocaleSummary], console: Console, dry_run: bool = False
+):
+    """
+    Remove extra translations that are not present in English.
+
+    This removes keys that exist in the target language but not in English.
+    """
+    for filename, diff in summary.items():
+        extra_keys = set(diff.extra_keys.get(language, []))
+        if not extra_keys:
+            continue
+
+        lang_path = LOCALES_DIR / language / filename
+        if not lang_path.exists():
+            continue
+
+        try:
+            lang_data = load_json(lang_path)
+        except Exception as e:
+            console.print(f"[red]Failed to load {language} file {lang_path}: {e}[/red]")
+            continue
+
+        # Helper to recursively remove extra keys
+        def remove_keys(data, prefix=""):
```

Comment:
```suggestion
        def remove_keys(data: dict[str, Any], prefix: str=""):
```

adding types for the parameters would help